Automattic/wp-calypso
# Happychat: Middleware refactor of HAPPYCHAT_SEND_MESSAGE

Creates new action type `HAPPYCHAT_SEND_MESSAGE` to move side-effects into middleware. Part of an ongoing refactor. Also adds first tests for HC reducers.

`HAPPYCHAT_SEND_MESSAGE`'s primary function is to send the message to the remote operator. But it also needs to inform the operator that typing has stopped, and update Redux state in `happychat.message` to clear the message text box.

### To test

- In side-by-side windows pull up the Happychat HUD and http://calypso.localhost:3000/me/chat
- Start typing in the message box in Calypso, and send a message
- The Calypso chat box should clear as soon as the message is sent
- The HUD operator should receive the message and see the "typing" animation disappear at the same time
